hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -95,6 +97,18 @@ default void init(@Nonnull Context context) throws Exception {
     @Nonnull
     Function<? super Address, ? extends ProcessorSupplier> get(@Nonnull List<Address> addresses);
 
+    /**
+     * Called on coordinator member after execution has finished on all
+     * members, successfully or not. The purpose is to figure out which
+     * {@link com.hazelcast.jet.Observable}s are being populated by the
+     * processors supplied so that completion or error events can be
+     * generated for those {@link com.hazelcast.jet.Observable}s.
+     */
+    @Nonnull
+    default Set<String> ownedObservables() {
```

Comment:
I don't like that this has leaked to PMS. This is really only implemented one processor?

Reply:
Two, but yeah, not nice. I had no better solution at the time. Will give finding one another try.

Reply:
This has been solved, PMS is no longer affected by this functionality.